eks/mng/scale/scale.go

```diff
@@ -0,0 +1,239 @@
+
+	if ts.cfg.EKSConfig.LogColor {
+		colorstring.Printf("\n\n[yellow]*********************************[default]\n")
+		colorstring.Printf("[light_green]MNGs[%q].Update[default]\n", mngName)
+	} else {
+		fmt.Printf("\n\n*********************************\n")
+		fmt.Printf("MNGs[%q].Update\n", mngName)
+	}
+
+	ts.cfg.Logger.Info("starting tester.Update", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	update.Created = true
+	ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName] = cur
+	cur, _ = ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName]
+	ts.cfg.EKSConfig.Sync()
+
+	createStart := time.Now()
+	defer func() {
+		createEnd := time.Now()
+		update.TimeFrameCreate = timeutil.NewTimeFrame(createStart, createEnd)
+		ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName] = cur
+		ts.cfg.EKSConfig.Sync()
+	}()
+
+	ts.cfg.Logger.Info("waiting before starting MNG update",
+		zap.String("cluster-name", ts.cfg.EKSConfig.Name),
+		zap.String("mng-name", mngName),
+		zap.Duration("initial-wait", update.InitialWait),
+	)
+	select {
+	case <-time.After(update.InitialWait):
+		ts.cfg.Logger.Info("waited, starting MNG scaler",
+			zap.String("cluster-name", ts.cfg.EKSConfig.Name),
+			zap.String("mng-name", mngName),
+			zap.Int("asg-min-size", cur.ASGMinSize),
+			zap.Int("asg-max-size", cur.ASGMaxSize),
+			zap.Int("asg-desired-capacity", cur.ASGDesiredCapacity),
+			zap.Int64("target-min-size", update.MinSize),
+			zap.Int64("target-max-size", update.MaxSize),
+			zap.Int64("target-desired-size", update.DesiredSize),
+		)
+	case <-ts.cfg.Stopc:
+		ts.cfg.Logger.Warn("MNG scaler wait aborted; exiting", zap.String("mng-name", mngName))
+		return errors.New("MNG scaler wait aborted")
+	}
+
+	// ref. https://docs.aws.amazon.com/cli/latest/reference/eks/update-nodegroup-config.html
+	var scaleErr = ts.scaleMNG(mngName, update)
+	if scaleErr != nil {
+		return scaleErr
+	}
+
+	ts.cfg.Logger.Info("completed MNG scaler",
+		zap.String("from", ts.cfg.EKSConfig.Parameters.Version),
+		zap.String("to", ts.cfg.EKSConfig.AddOnClusterVersionUpgrade.Version),
+	)
+	return nil
+}
+
+// Scales MNG to the next scaling configuration in the array.
+func (ts *tester) scaleMNG(mngName string, update *eksconfig.MNGScaleUpdate) (err error) {
+	cur := ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName] 
+	nodegroupConfigInput := &aws_eks.UpdateNodegroupConfigInput{
+		ClusterName:   aws.String(ts.cfg.EKSConfig.Name),
+		NodegroupName: aws.String(mngName),
+		ScalingConfig: &aws_eks.NodegroupScalingConfig{DesiredSize: aws.Int64(update.DesiredSize),
+			MaxSize: aws.Int64(update.MaxSize),
+			MinSize: aws.Int64(update.MinSize)}}
+	var updateOut *eks.UpdateNodegroupConfigOutput
+	updateOut, err = ts.cfg.EKSAPI.UpdateNodegroupConfig(nodegroupConfigInput)
+
+	if err != nil {
+		ts.cfg.Logger.Warn("MNG scaler request failed", zap.String("mng-name", mngName), zap.Error(err))
+		return err
+	}
+	if updateOut.Update != nil {
+		reqID = aws.StringValue(updateOut.Update.Id)
```

Comment:
Let's not use global variable `reqID` (which could be racey if this method runs in parallel)